src/test/java/org/alliancegenome/curation_api/ExpressionBulkUploadFmsITCase.java

```diff
@@ -0,0 +1,93 @@
+package org.alliancegenome.curation_api;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.restassured.RestAssured;
+import io.restassured.config.HttpClientConfig;
+import io.restassured.config.RestAssuredConfig;
+import org.alliancegenome.curation_api.base.BaseITCase;
+import org.alliancegenome.curation_api.constants.VocabularyConstants;
+import org.alliancegenome.curation_api.model.entities.DataProvider;
+import org.alliancegenome.curation_api.model.entities.ResourceDescriptor;
+import org.alliancegenome.curation_api.model.entities.Vocabulary;
+import org.alliancegenome.curation_api.model.entities.VocabularyTerm;
+import org.alliancegenome.curation_api.resources.TestContainerResource;
+import org.junit.jupiter.api.*;
+
+import java.util.List;
+
+
+@QuarkusIntegrationTest
+@QuarkusTestResource(TestContainerResource.Initializer.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@DisplayName("606 - Expression bulk upload - FMS")
+@Order(606)
+public class ExpressionBulkUploadFmsITCase extends BaseITCase {
+
+	private final String expressionBulkPostEndpoint = "/api/gene-expression-annotation/bulk/ZFIN/annotationFile";
+	private final String expressionTestFilePath = "src/test/resources/bulk/fms/07_expression/";
+	private final String expressionFindEndpoint = "/api/gene-expression-annotation/find?limit=100&page=0";
+	private final String taxon = "NCBITaxon:7955";
+	private final String gene = "GEXPTEST:GENE001";
+	private final String mmoTerm = "GEXPTEST:assay001";
+	private final String referenceId = "ZFIN:PUB001";
+	private final String publicationId = "PMID:009";
+
+	@BeforeEach
+	public void init() {
+		RestAssured.config = RestAssuredConfig.config()
+			.httpClient(HttpClientConfig.httpClientConfig()
+				.setParam("http.socket.timeout", 100000)
+				.setParam("http.connection.timeout", 100000));
+	}
+
+	@Test
+	@Order(1)
+	public void expressionBulkUploadAllFields() throws Exception {
+		loadRequiredEntities();
+		checkSuccessfulBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "AF_01_all_fields.json");
+	}
+
+	@Test
+	@Order(2)
+	public void expressionBulkUploadMissingRequiredFields() throws Exception {
+		checkFailedBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "MR_01_no_geneId.json");
+		checkFailedBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "MR_02_no_dateAssigned.json");
+		checkFailedBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "MR_03_no_evidence.json");
+		checkFailedBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "MR_04_no_assay.json");
+		checkFailedBulkLoad(expressionBulkPostEndpoint, expressionTestFilePath + "MR_05_no_whenExpressed.json");
```

Comment:
added for reference and publication id

Reply:
This still needs specific field tests.  For example, your test file for missing whenExpressed is missing the entire whenExpressed object.  What we really need to test for is each of the required fields - your file/test doesn't test whether the code correctly deals with an entry that has a whenExpressed object but that is missing of of the required fields within that object.  I'm fairly sure your code does deal with this correctly, but the test needs to capture that so that any future code changes don't have unintended consequences.